frontend/composables/use-group-recipe-actions.ts

```diff
@@ -46,8 +45,40 @@ export const useGroupRecipeActions = function (
     return groupRecipeActions.value;
   });
 
-  function parseRecipeActionUrl(url: string, recipe: Recipe): string {
+  function getTokenLink(token: string, groupSlug: string) {
+    return `${window.location.origin}/g/${groupSlug}/shared/r/${token}`;
+  }
+
+  async function parseRecipeActionUrl(url: string, recipe: Recipe): Promise<string> {
     /* eslint-disable no-template-curly-in-string */
+    const shareLinkRegex = /\$\{share-link-expires-seconds-[0-9]+\}/g;
+    const group = (await api.groups.getOne(recipe.groupId || "")).data;
```

Comment:
You should be able to grab the slug from the URL instead of fetching it via the API:

```js
import { useRoute } from "@nuxtjs/composition-api";

// put this right under "useUserApi"
const route = useRoute();

// replace the API call with this
const groupSlug = route.value.params.groupSlug || recipe.groupId || "";
```